From d0128287a671f1a009b8062e3e9bfa877b36be8c Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 30 Mar 2011 01:45:17 +0200
Subject: [PATCH] base/dracut-lib.sh: relax getargbool value parsing

if a non-boolean value was specified, like "yes" or "no",
getargbool() would error, because we check for "$val -eq 0"
---
 modules.d/99base/dracut-lib.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/modules.d/99base/dracut-lib.sh b/modules.d/99base/dracut-lib.sh
index a5406ab..37b3e76 100755
--- a/modules.d/99base/dracut-lib.sh
+++ b/modules.d/99base/dracut-lib.sh
@@ -98,7 +98,7 @@ getargbool() {
     _b=$(getarg "$@")
     [ $? -ne 0 -a -z "$_b" ] && _b=$_default
     if [ -n "$_b" ]; then
-        [ $_b -eq 0 ] && return 1
+        [ $_b = "0" ] && return 1
         [ $_b = "no" ] && return 1
     fi
     return 0
